`azurerm_application_gateway` - support for redirect rules (#2908)
terraform-providers/terraform-provider-azurerm: website/docs/r/application_gateway.html.markdown

* `custom_error_page_url` - (Required) Error page URL of the application gateway customer error.

---

A `redirect_configuration` block supports the following:

* `name` - (Required) Unique name of the redirect configuration block

* `redirect_type` - (Required) The type of redirect. Possible values are `Permanent`, `Temporary`, `Found` and `SeeOther`

> * `target_listener_name` - (Optional) The name of the listener to redirect to. This parameter is mutually exclusive with `target_url`.

Reviewer: for consistency with other resources, can we make this:

```suggestion
* `target_listener_name` - (Optional) The name of the listener to redirect to. This field must be set when `target_url` is set.
```

Author: Yes, sure, I assume you meant "field must NOT be set when `target_url` is set" ;)  It's a mutual exclusion.

Author: Done. I made the changes from my branch

> * `target_url` - (Optional) The Url to redirect the request to. This parameter is mutually exclusive with `target_listener_name`.

Reviewer: for consistency with other resources, can we make this:

```suggestion
* `target_url` - (Optional) The Url to redirect the request to. This field must be set when `target_listener_name` is set.
```

Author: Done
